CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+
+## Requirements
+
+- WordPress 5.0+ or the [Gutenberg Plugin](https://wordpress.org/plugins/gutenberg/).
+- PHP 5.6.20 or greater, [Composer](https://getcomposer.org) and [Node.js](https://nodejs.org) for dependency management.
+- [Docker](https://docs.docker.com/install/) for a local development environment.
+
+We suggest using a software package manager for installing the development dependencies such as [Homebrew](https://brew.sh) on MacOS:
+
+	brew install php composer node docker docker-compose
+
+or [Chocolatey](https://chocolatey.org) for Windows:
+
+	choco install php composer node nodejs docker-compose
+
+## Issue Reporting Guidelines
+
+- The issue list of this repo is **exclusively** for bug reports and feature requests.
+- Try to search for your issue, it may have already been answered or even fixed in the `develop` branch.
+- Check if the issue is reproducible with the latest stable version. If you are using a pre-release, please indicate the specific version you are using.
+- It is **required** that you clearly describe the steps necessary to reproduce the issue you are running into. Issues without clear reproducible steps will be closed immediately.
+- If your issue is resolved but still open, don't hesitate to close it. In case you found a solution by yourself, it could be helpful to explain how you fixed it.
+
+## Pull Request Guidelines
+
+- Checkout a topic branch from `develop` and merge back against `develop`.
+    - If you are not familiar with branching please read [_A successful Git branching model_](http://nvie.com/posts/a-successful-git-branching-model/) before you go any further.
+- Follow the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/coding-standards/).
+- Make sure the `phpcs` and `phpunit` composer scripts pass. (see [development setup](#development-setup))
+- If adding a new feature:
+    - Add accompanying test case.
+    - Provide convincing reason to add this feature. Ideally you should open a suggestion issue first and have it green-lit before working on it.
+- If fixing a bug:
+    - Provide detailed description of the bug in the PR. Live demo preferred.
+    - Add appropriate test coverage if applicable.
+
+## Continuous Integration
+
+We use [Travis CI](https://travis-ci.com) to lint all code, run tests and report test coverage to [Coveralls](https://coveralls.io) as defined in [`.travis.yml`](.travis.yml). Travis CI will run the unit tests and perform sniffs against the WordPress Coding Standards whenever you push changes to your PR. Tests are required to pass successfully for a merge to be considered.
+
+## Development
+
+1. Clone the plugin repository.
+
+	    git clone git@github.com:xwp/wp-foo-bar.git
+
+2. Setup the development tools using [Composer](https://getcomposer.org):
+
+	    composer install
+
+    This will automatically install the `pre-commit` hook from the `wp-dev-lib` Composer package and setup the unit tests.
+
```

Comment:
@derekherman: 'npm' install  would need to be added as a 3rd step right for the people contributing (i.e not running init-plugin.sh)?

Reply:
@davidlonjon Good catch, I changed things to wrap everything with `npm` but missed that comment. I'll fix it now. Cheers!